### DISCUSSION TITLE
Support for tracker module audio files (such as .mod, .it, and .xm)

### DISCUSSION THREAD

**Discussion by FazlyMR** [ppy/osu-framework#5651] [Ideas] [answered]
Considering the BASS audio library already supports, as mentioned in it's [homepage as one of it's main features](https://www.un4seen.com/bass.html) (MOD music playback), the decoding (and playback) of [tracker module files](https://en.wikipedia.org/wiki/Module_file), it would be nice to have support for it in osu!framework.

While osu!lazer nor osu!framework does not state official support for such files for use in beatmaps and playback, respectively (so this error is expected). Currently, as tested in osu!lazer version 2023.207.0.0, with osu!framework as one of it's dependencies, playback of such files are not supported and forcing it to load such files results in osu!framework errors.

**Comment by peppy** [comment, marked as answer]
If you're asking for support specifically for use in the osu! ecosystem, I do not foresee this happening. It would need a lot of community support, and would need to be implemented in a way to actually benefit from the tracker format (ie. to provide hitsounds). Also consider that it would not work with third party tools, and would require infrastructure support to generate previews etc.

Quite a large undertaking, and given no one has shown interest in 15 years I don't think it's a good time investment.

If you're asking for your own project, I think it will be up to you to implement yourself. Whether we'd want that as part of o!f or not depends on how much is involved (I'd hope it was only a few support classes at most).

We don't seal most classes in o!f so you could likely implement this locally in your own project in a way that worked well without requiring anything from us.
